File: src/main/java/com/fifo/ticketing/domain/performance/dto/LikedPerformanceDto.java
package com.fifo.ticketing.domain.performance.dto;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;


@Getter
@Builder
@AllArgsConstructor
public class LikedPerformanceDto {

    private Long id;
    private String title;
    private String encodedFileName;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String placeName;
    private String urlPrefix;

    public String getUrl() {
        return urlPrefix + encodedFileName;
    }
}

File: src/main/java/com/fifo/ticketing/domain/performance/dto/LikedPerformanceDto.kt
package com.fifo.ticketing.domain.performance.dto

import java.time.LocalDateTime

data class LikedPerformanceDto(
    val id: Long,
    val title: String,
    val encodedFileName: String,
    val startTime: LocalDateTime,
    val endTime: LocalDateTime?,
    val placeName: String,
    val urlPrefix: String,
) {
    val url: String
        get() = urlPrefix + encodedFileName
}
